src/metabase/lib/field.cljc

```diff
@@ -17,12 +18,13 @@
   field-clause)
 
 (defmethod ->field :metadata/field
-  [_query _stage-number field-metadata]
-  (lib.options/ensure-uuid
-   (or (:field_ref field-metadata)
-       (when-let [id (:id field-metadata)]
-         [:field id nil])
-       [:field (:name field-metadata) {:base-type (:base_type field-metadata)}])))
+  [_query _stage-number {base-type :base_type, field-id :id, field-name :name, field-ref :field_ref, :as _field-metadata}]
+  (cond-> (or (when field-ref
+                (lib.convert/->pMBQL field-ref))
+              [:field {} (or field-id
+                             field-name)])
+    base-type (lib.options/update-options assoc :base-type base-type)
```

Comment:
Include `:base-type` in `:field` clause options whenever it's available, as discussed on Slack. This will let us do strict type checking in the Malli schema itself in the future